dts/bindings/base/mutable.yaml

```diff
@@ -0,0 +1,13 @@
+# Copyright (c) 2023, Meta
+# SPDX-License-Identifier: Apache-2.0
+
+# Properties for Mutable devices
+
+properties:
+  zephyr,mutable:
```

Comment:
What's the point of this property.  If `zephyr,devmux` is the only thing that sets it why is this needed?

Reply:
Technically, any device could set this property but it's really only useful for devices that need to have a runtime modifiable interface for whatever reason.

There are many possible use cases. Some identified in the earlier PR comments, so the utility shouldn't be limited to `zephyr,devmux`.

Reply:
Does it make sense to add it later?

Is this more descriptive of what it's meaning:

`zephyr,device-api-mutuable`?

Reply:
I'm ok with the name change, but the property is definitely needed for devmux to work correctly / not fault.

Reply:
I think `zephyr,mutable` is just fine, short and clear.

Reply:
@galak, @gmarull - I'm ok with naming things either way, but my preference is the short + sweet version.

Reply:
to me PR is ok as is

Reply:
@galak - I can use your suggestion of `zephyr,device-api-mutuable`. @gmarull - does it matter to you? Would likely need to rebase anyway and get approvals again.